File: benchmarks/competitors/bench_loop.go
// ==============================================================================
// 🐹 GO BENCHMARK: 10,000,000 ARITHMETIC LOOP ITERATIONS (benchmarks/bench_loop.go)
// ==============================================================================

package main

import (
	"fmt"
	"time"
)

func main() {
	start := time.Now()

	var total uint64 = 0
	for i := uint64(1); i <= 10000000; i++ {
		total += (i % 7)
	}

	elapsed := time.Since(start)

	fmt.Println("Go Loop 10,000,000 Iterations Complete!")
	fmt.Printf("Sum: %d\n", total)
	fmt.Printf("Elapsed Time: %.2f ms\n", float64(elapsed.Microseconds())/1000.0)
}

File: benchmarks/competitors/bench_loop.swift
// ==============================================================================
// 🕊️ SWIFT BENCHMARK: 10,000,000 ARITHMETIC LOOP ITERATIONS (benchmarks/bench_loop.swift)
// ==============================================================================

import Foundation
import Dispatch

let start = DispatchTime.now().uptimeNanoseconds

var total: UInt64 = 0
for i: UInt64 in 1...10_000_000 {
    total &+= (i % 7)
}

let end = DispatchTime.now().uptimeNanoseconds
let elapsed_ms = Double(end - start) / 1_000_000.0

print("Swift Loop 10,000,000 Iterations Complete!")
print("Sum: \(total)")
print(String(format: "Elapsed Time: %.2f ms", elapsed_ms))
